### ISSUE TITLE
Can't install root on ubuntu 23

### ISSUE THREAD

**Issue by minimeu1** [root-project/root#14479]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

Hello! Sorry for the newbie issue, but I can't seem to work around it...!
After upgrading from ubuntu 22 to ubuntu 23, my root stopped working. I would try to run my old macros and the system would always break (can't find those logs now, I'm sorry!)

### Reproducer

I was using the precompiled version and therefore I have uninstalled it and tried to reinstall it again. From the root webpage I was suggested to do so via snap. I managed to install it and in the end, when I oppened a session and tried new TBrowser() to check some hists and if everything was according to how it was previously, I got this as header of the break:

```
cling::DynamicLibraryManager::loadLibrary(): libAfterImage.so.0: cannot open shared object file: No such file or directory
Error in <TInterpreter::TCling::AutoLoad>: failure loading library libASImage.so for TASImage
cling::DynamicLibraryManager::loadLibrary(): libAfterImage.so.0: cannot open shared object file: No such file or directory
Error in <TGVSplitter::TGVSplitter>: splitterv.xpm not found
Error in <TGHSplitter::TGHSplitter>: splitterh.xpm not found
Error in <TGComboBox::TGComboBox>: arrow_down.xpm not found
Error in <TGVScrollBar::TGVScrollBar>: arrow_*.xpm not found
```

I proceeded to uninstall it and try to build it on my own.
I followed the instructions on this webpage https://root.cern/install/build_from_source/ after checking all the dependencies. When I made the first cmake I got some things that it couldn't find which would give the following lines on terminal

```
Could NOT find Vdt (missing: VDT_INCLUDE_DIR VDT_LIBRARY) (Required is at least version "0.4")
-- VDT not found. Ensure that the installation of VDT is in the CMAKE_PREFIX_PATH
```
The things it couldn't find were:
```
PCRE
xxHash
ZSTD
GIF
TIFF
JPEG
GLEW
gl2ps
LibXml2
MySQL
PostgreSQL
Sqlite
CFITSIO
XROOTD
FTGL
LibXml2
TBB
Vdt
OCaml
```
Then I got the following lines 
```
-- clang project is enabled
-- clang-tools-extra project is disabled
-- compiler-rt project is disabled
-- cross-project-tests project is disabled
-- libc project is disabled
-- libclc project is disabled
-- libcxx project is disabled
-- libcxxabi project is disabled
-- libunwind project is disabled
-- lld project is disabled
-- lldb project is disabled
-- mlir project is disabled
-- openmp project is disabled
-- parallel-libs project is disabled
-- polly project is disabled
-- pstl project is disabled
-- flang project is disabled
```
and these:
```
-- LLVM host triple: x86_64-unknown-linux-gnu
-- LLVM default target triple: x86_64-unknown-linux-gnu
-- Building with -fPIC
-- Found Python3: /usr/bin/python3 (found suitable version "3.11.6", minimum required is "3.0") found components: Interpreter 
-- Linker detection: GNU ld
-- Targeting X86
-- Targeting NVPTX
-- Clang version: 13.0.0
-- Not building amdgpu-arch: hsa-runtime64 not found
-- Building Cling as part of ROOT
-- Cling version (from VERSION file): ROOT_1.0~dev
-- Cling will look for C++ headers in '/usr/include/c++/13:/usr/include/x86_64-linux-gnu/c++/13:/usr/include/c++/13/backward' at runtime.
-- And then fallback to: 'x86_64-linux-gnu-g++-13'
-- Performing Test CXX_HAS_fno_rtti
```

The last lines of this cmake were:
```cmake
-- ROOT Configuration 

System          Linux-6.5.0-15-generic
Processor       2 core Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz (x86_64)
Build type      Release
Install path    /home/<user>/root_install
Compiler        GNU 13.2.0
Compiler flags:
C                -Wno-implicit-fallthrough -pipe -Wall -W -pthread -O3 -DNDEBUG
C++              -std=c++17 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe  -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -pthread -O3 -DNDEBUG
Linker flags:
Executable       -rdynamic
Module          
Shared           -Wl,--no-undefined -Wl,--hash-style="both"

-- Enabled support for:  asimage builtin_afterimage builtin_clang builtin_cling builtin_ftgl builtin_gl2ps builtin_glew builtin_llvm builtin_lz4 builtin_lzma builtin_nlohmannjson builtin_openui5 builtin_pcre builtin_tbb builtin_vdt builtin_xrootd builtin_xxhash builtin_zstd clad dataframe gdml http imt mlp opengl proof pyroot roofit webgui root7 rpath runtime_cxxmodules shared ssl tmva tmva-cpu tmva-pymva spectrum vdt x11 xrootd
-- Configuring done (28.0s)
-- Generating done (3.2s)
-- Build files have been written to: /home/<user>
```

I tried to run `cmake --build . --target install` and ended up with

```
CMake Error: The current CMakeCache.txt directory /home/<user>/root_build/CMakeCache.txt is different than the directory /home/<user> where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
CMake Error: Target DependInfo.cmake file not found
Creating directories for 'LZMA'
CMake Error: Error processing file: /home/francisco/LZMA-prefix/tmp/LZMA-mkdirs.cmake
gmake[2]: *** [CMakeFiles/LZMA.dir/build.make:108: LZMA-prefix/src/LZMA-stamp/LZMA-mkdir] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:8564: CMakeFiles/LZMA.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
``` 

### ROOT version

supposedly the most recent 
root_v6.30.02

### Installation method

build from source

### Operating system

Ubuntu 23.10 

### Additional context

I have been using Ubuntu and ROOT for some years now but when I come across this type of problems normally I can work them and come up with a solution. But all this with installations and building from source is quite new to me. When I have done it before I have never came across problems that would put me reading through the outputs.
I am sorry if it definetly not contextualized but please guide me through this issue :)
Thank you in advance for your help!

**Comment by dpiparo** [comment]
Hi @minimeu1 ,
a new fresh build, starting from clean directories seems required. First step: did you make sure all the dependencies for kubuntu are installed? https://root.cern/install/dependencies/#ubuntu-and-other-debian-based-distributions

**Comment by ferdymercury** [comment]
> cling::DynamicLibraryManager::loadLibrary(): libAfterImage.so.0: cannot open shared object file: No such file or directory
> Error in <TInterpreter::TCling::AutoLoad>: failure loading library libASImage.so for TASImage
> cling::DynamicLibraryManager::loadLibrary(): libAfterImage.so.0: cannot open shared object file: No such file or directory

A workaround to solve this is to do sudo apt install libafterimage0-dev, then it should work with the precompiled binary.

(In the meantime, this issue will be solved in the next release, see https://github.com/root-project/root/pull/14402)

So for the moment, I would suggest to try to go back to the precompiled binary and install that missing dependency.

**Comment by ferdymercury** [comment]
> Could NOT find Vdt (missing: VDT_INCLUDE_DIR VDT_LIBRARY) (Required is at least version "0.4")
> -- VDT not found. Ensure that the installation of VDT is in the CMAKE_PREFIX_PATH

This seems a bit similar to what will be fixed in the next release, by @krasznaa or @bellenot 

https://github.com/root-project/root/issues/14163
https://github.com/root-project/root/pull/14474

**Comment by minimeu1** [comment]
Hello both of you!
Thank you so much for the quick reply!
I went through my text again and didn't notice it repeated itself...! Sorry for that!

Considering the suggestion from @dpiparo , I have checked all the dependencies. Some I had to install, but now they are all in the system. I have tried to go through the cmake again and got the same problems. You suggest me to try from scratch again. I can do this simply by deleting the folders I have created for the installation and redo the process in the page "build from source"?

I have tried the workaround suggested by @ferdymercury ! Firstly I have tried the snap version and nothing happened. Same errors. Now I have downloaded the pre-compiled version and I can use it, open canvases and browsers with no problem. I will run some macros to compare with the outputs I had previously and check if everything is according to what I had.

Eventhough it is already working, I am willing to try and build it from source if that can help you somehow figuring out some kind of challanges you would like to improve in further versions! Please let me know if I can be of any use!

Thank you very much once again! It is always amazing to receive this kind of comprehensible support :))))

**Comment by ferdymercury** [comment]
> Eventhough it is already working, I am willing to try and build it from source if that can help you somehow figuring out some kind of challanges you would like to improve in further versions! Please let me know if I can be of any use!

Yes, in that case if you want to help with testing, please build from source, but do not download the source as .tar.gz, but clone instead the github repository and checkout the 6.30.00-patches branch, or the master branch. This one will include the fixes I mentioned.

**Comment by ferdymercury** [comment]
This should be fixed with yesterday's release of 6.30.04
Please reopen if you find still an issue ;)
